lib/wit/deps/fastly/compute.wit

```diff
@@ -0,0 +1,1076 @@
+    generic-error,
+    /// Invalid argument.
+    invalid-argument,
+    /// Invalid handle.
+    /// Thrown when a handle is not valid. E.G. No dictionary exists with the given name.
+    bad-handle,
+    /// Buffer length error.
+    /// Thrown when a buffer is too long.
+    buffer-len,
+    /// Unsupported operation error.
+    /// This error is thrown when some operation cannot be performed, because it is not supported.
+    unsupported,
+    /// Alignment error.
+    /// This is thrown when a pointer does not point to a properly aligned slice of memory.
+    bad-align,
+    /// Invalid HTTP error.
+    /// This can be thrown when a method, URI, header, or status is not valid. This can also
+    /// be thrown if a message head is too large.
+    http-invalid,
+    /// HTTP user error.
+    /// This is thrown in cases where user code caused an HTTP error. For example, attempt to send
+    /// a 1xx response code, or a request with a non-absolute URI. This can also be caused by
+    /// an unexpected header: both `content-length` and `transfer-encoding`, for example.
+    http-user,
+    /// HTTP incomplete message error.
+    /// This can be thrown when a stream ended unexpectedly.
+    http-incomplete,
+    /// A `None` error.
+    /// This status code is used to indicate when an optional value did not exist, as opposed to
+    /// an empty value.
+    /// Note, this value should no longer be used, as we have explicit optional types now.
+    optional-none,
+    /// Message head too large.
+    http-head-too-large,
+    /// Invalid HTTP status.
+    http-invalid-status,
+    /// Limit exceeded
+    ///
+    /// This is returned when an attempt to allocate a resource has exceeded the maximum number of
+    /// resources permitted. For example, creating too many response handles.
+    limit-exceeded
+  }
+
+  type secret-handle = u32;
+}
+
+interface fastly-abi {
+  use types.{error};
+  init: func(version: u64) -> result<_, error>;
+}
+
+interface http-types {
+
+  use types.{secret-handle};
+
+  type body-handle = u32;
+
+  type request-handle = u32;
+  type pending-request-handle = u32;
+  type response-handle = u32;
+  type request = tuple<request-handle, body-handle>;
+  type response = tuple<response-handle, body-handle>;
+
+  enum http-version {
+    http09,
+    http10,
+    http11,
+    h2,
+    h3
+  }
+
+  flags content-encodings {
+    gzip
+  }
+
+  /// Adjust how this requests's framing headers are determined.
+  enum framing-headers-mode {
+    automatic,
+    manually-from-headers
+  }
+
+  enum tls-version {
+    tls1,
+    tls11,
+    tls12,
+    tls13
+  }
+
+  flags backend-config-options {
+    reserved,
+    host-override,
+    connect-timeout,
+    first-byte-timeout,
+    between-bytes-timeout,
+    use-ssl,
+    ssl-min-version,
+    ssl-max-version,
+    cert-hostname,
+    ca-cert,
+    ciphers,
+    sni-hostname,
+    dont-pool,
+    client-cert,
+    grpc,
+  }
```

Comment:
Do flags get encoded in `wit` the same way they do in `witx`; i.e., this is a i32 with each of these items being bits, defined in the provided order?

We did that as a way to make the ABI stable under expansion, although it looks like there are other ways to achieve versioning in `wit`. But depending on how this reaches the ABI and/or the Fastly SDKs, I can see trouble looming in this sort of thing.

Reply:
Yep, they get compiled in much the same way. Wit has better features for versioning that we should lean on in the future, but in the meantime I've converted by individually translating each flag in a `From` impl :(

Reply:
Oh that sounds like it was a joy and a privilege. Thank you for your service!